### PR TITLE
More api compatibility doc cleanup

### PR DESCRIPTION
**Pull request by bparees**
Change is applicable to 4.10+

### CODE REVIEW

**Comment by bparees** [comment]
@derekwaynecarr ptal.  I missed a few spots when i updated the tier 1 compatibility commitment to aligh w/ k8s, but also a few other spots can benefit from some clarification, i think.

File: modules/api-compatibility-guidelines.adoc [hunk 1]
```diff
@@ -13,6 +13,7 @@ Red Hat recommends that application developers adopt the following principles in
 * Do not design applications that rely on configuration files provided by system packages or other components. These files can change between versions unless the upstream community is explicitly committed to preserving them. Where appropriate, depend on any Red Hat provided interface abstraction over those configuration files in order to maintain forward compatibility. Direct file system modification of configuration files is discouraged, and users are strongly encouraged to integrate with an Operator provided API where available to avoid dual-writer conflicts.
 * Do not depend on API fields prefixed with `unsupported<FieldName>` or annotations that are not explicitly mentioned in product documentation.
 * Do not depend on components with shorter compatibility guarantees than your application.
+* Do not interact directly with the etcd server, all etcd access must be performed via the api-server.
```
Comment (sferich888) [inline modules/api-compatibility-guidelines.adoc:16]: Even for backups and restores operations? 

IE: https://docs.openshift.com/container-platform/4.9/backup_and_restore/index.html

I assume this is excluded; but 'direct access' or 'interaction' is required for these procedures.

Comment (bparees) [reply modules/api-compatibility-guidelines.adoc:16]: good catch, no the thing we're trying to exclude here is people making direct create/update calls to store their own objects in etcd (or, worse, updating ours)

Comment (bparees) [reply modules/api-compatibility-guidelines.adoc]: @sferich888 updated, see if that looks better

Comment (bparees) [reply modules/api-compatibility-guidelines.adoc]: (note that i've shoehorned this into the "application developer" guidance.  i didn't see a better place to document this and i didn't want to create a whole new one for this item)

**Review by derekwaynecarr — APPROVED** [review]
/lgtm
/approve

**Comment by bparees** [comment]
@openshift/team-documentation ptal

File: modules/api-compatibility-guidelines.adoc [hunk 2]
```diff
@@ -13,6 +13,7 @@ Red Hat recommends that application developers adopt the following principles in
 * Do not design applications that rely on configuration files provided by system packages or other components. These files can change between versions unless the upstream community is explicitly committed to preserving them. Where appropriate, depend on any Red Hat provided interface abstraction over those configuration files in order to maintain forward compatibility. Direct file system modification of configuration files is discouraged, and users are strongly encouraged to integrate with an Operator provided API where available to avoid dual-writer conflicts.
 * Do not depend on API fields prefixed with `unsupported<FieldName>` or annotations that are not explicitly mentioned in product documentation.
 * Do not depend on components with shorter compatibility guarantees than your application.
+* Do not perform direct storage operations on the etcd server, all etcd access must be performed via the api-server or through documented backup/restore procedures.
```
Comment (sagidlow) [inline modules/api-compatibility-guidelines.adoc:16]:
```suggestion
* Do not perform direct storage operations on the etcd server, all etcd access must be performed via the api-server or through documented backup or restore procedures.
```
IBM Style Guide: Avoid using forward slashes (/) and backslashes (\) in most text and in dates.

**Review by sagidlow — COMMENTED** [review]
One nit comment to fix.

**Comment by sagidlow** [comment]
For future reference:

- In your original comment, please post which releases this change applies to i.e., 4.6+ 
- Please post a request for review on the #forum-docs-review channel on Slack.

**Comment by bparees** [comment]
@sagidlow thanks, fixed.  The change should go into 4.10+


```
For future reference:

    In your original comment, please post which releases this change applies to i.e., 4.6+
    Please post a request for review on the #forum-docs-review channel on Slack.
```

If this is the expectation, please consider setting up automation created that informs PR authors of these steps.  I'm not going to promise to remember the rules for how to engage w/ the docs team when i submit a docs PR once every few months.

File: modules/api-compatibility-guidelines.adoc [hunk 3]
```diff
@@ -13,6 +13,7 @@ Red Hat recommends that application developers adopt the following principles in
 * Do not design applications that rely on configuration files provided by system packages or other components. These files can change between versions unless the upstream community is explicitly committed to preserving them. Where appropriate, depend on any Red Hat provided interface abstraction over those configuration files in order to maintain forward compatibility. Direct file system modification of configuration files is discouraged, and users are strongly encouraged to integrate with an Operator provided API where available to avoid dual-writer conflicts.
 * Do not depend on API fields prefixed with `unsupported<FieldName>` or annotations that are not explicitly mentioned in product documentation.
 * Do not depend on components with shorter compatibility guarantees than your application.
+* Do not perform direct storage operations on the etcd server, all etcd access must be performed via the api-server or through documented backup and restore procedures.
```
Comment (stevsmit) [inline modules/api-compatibility-guidelines.adoc:16]: This is a run-on sentence. There should be a period or conjunction (and) after 'on the etcd server'. I prefer a period!

File: modules/api-support-tiers-mapping.adoc [hunk 4]
```diff
@@ -5,7 +5,7 @@
 [id="api-support-tiers-mapping_{context}"]
 = Mapping API tiers to API groups
 
-For each API tier defined by Red Hat, we provide a mapping table for specific API groups where the upstream communities are committed to maintain forward compatibility. Any API group that does not specify an explicit compatibility level is assigned API tier 3 by default.
+For each API tier defined by Red Hat, we provide a mapping table for specific API groups where the upstream communities are committed to maintain forward compatibility. Any API group that does not specify an explicit compatibility level and is not specifically discussed below is assigned API tier 3 by default except for v1alpha1 APIs which are assigned tier 4 by default.
```
Comment (stevsmit) [inline modules/api-support-tiers-mapping.adoc:8]: `v1alpha1` APIs should be in backticks.

File: modules/api-support-tiers.adoc [hunk 5]
```diff
@@ -29,4 +29,4 @@ Components and developer tools that receive continuous updates through the Opera
 == API tier 4
 No compatibility is provided. API and AOE can change at any point. These capabilities should not be used by applications needing long-term support.
 
-It is common practice for Operators to use custom resource definitions (CRDs) internally to accomplish a task.  These objects are not meant for use by actors external to the Operator and are intended to be hidden.  If any CRD is not meant for use by actors external to the Operator, the `operators.operatorframework.io/internal-objects` annotation in the Operators `ClusterServiceVersion` (CSV) must be specified, and that signals that the corresponding resource is internal use only.
+It is common practice for Operators to use custom resource definitions (CRDs) internally to accomplish a task.  These objects are not meant for use by actors external to the Operator and are intended to be hidden.  If any CRD is not meant for use by actors external to the Operator, the `operators.operatorframework.io/internal-objects` annotation in the Operators `ClusterServiceVersion` (CSV) should be specified to signal that the corresponding resource is internal use only and the CRD may be explicitly labeled as tier 4.
```
Comment (stevsmit) [inline modules/api-support-tiers.adoc:32]: There are two spaces after 'hidden.' and before 'If.' 

There are two spaces after 'task.' and before 'These'.

**Review by stevsmit — APPROVED** [review]

File: modules/api-compatibility-guidelines.adoc (hunk 3 above)
Comment (bparees) [reply modules/api-compatibility-guidelines.adoc]: fixed

File: modules/api-support-tiers-mapping.adoc (hunk 4 above)
Comment (bparees) [reply modules/api-support-tiers-mapping.adoc]: fixed

File: modules/api-support-tiers.adoc (hunk 5 above)
Comment (bparees) [reply modules/api-support-tiers.adoc]: fixed

**Pull request merged**